cmd/snap-update-ns/change.go

```diff
@@ -125,8 +125,8 @@ func (c *Change) createPath(path string, pokeHoles bool, as *Assumptions) ([]*Ch
 	if needsMimic, mimicPath := mimicRequired(err); needsMimic && pokeHoles {
 		// If the error can be recovered by using a writable mimic
 		// then construct one and try again.
-		logger.Debugf("need to create writable mimic needed to create path %q (original error: %v)", path, err)
-		changes, err = createWritableMimic(mimicPath, path, as)
+		logger.Debugf("need to create writable mimic needed to create path %q (mount entry id: %q) (original error: %v)", path, c.Entry.XSnapdEntryID(), err)
+		changes, err = createWritableMimic(mimicPath, c.Entry.XSnapdEntryID(), as)
```

Comment:
I do not see any code actually creating a "x-snapd.id=" option entry.
Unlike other options, `(e *MountEntry) XSnapdEntryID()`, does not have a counterpart function to generate the string. **It looks like an unused concept.. am I missing something?**

If this is the case, we are always identifying mounts by target (Dir), which is used to associate synthetics produced for target (Dir) or source (Name) originated mimics. Perhaps there is compatibility reason to keep `x-snapd.id`, otherwise we may want to remove it.

Reply:
yes, there is very little code that seems involved with it as a entry option, so it's currently just a synonym of MountEntry.Dir